cmd/skywire-cli/commands/log/root.go

```diff
@@ -210,13 +242,93 @@ func download(ctx context.Context, log *logging.Logger, httpC http.Client, targe
 	file, _ := os.Create(pubkey + "/" + fileName) //nolint
 	defer file.Close()                            //nolint
 
-	if err := dmsgget.Download(ctx, log, &httpC, file, target, maxSize); err != nil {
+	if err := Download(ctx, log, &httpC, file, target, maxSize); err != nil {
 		log.WithError(err).Errorf("The %s for visor %s not available", fileName, pubkey)
 		return err
 	}
 	return nil
 }
 
+// Download downloads a file from the given URL into 'w'.
+func Download(ctx context.Context, log logrus.FieldLogger, httpC *http.Client, w io.Writer, urlStr string, maxSize int64) error {
+	req, err := http.NewRequest(http.MethodGet, urlStr, nil)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to formulate HTTP request.")
+	}
+	resp, err := httpC.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to connect to HTTP server: %w", err)
+	}
+	if resp.StatusCode == http.StatusOK {
+		// 200 OK
+		if maxSize > 0 {
+			if resp.ContentLength > maxSize*1024 {
+				return fmt.Errorf("requested file size is more than allowed size: %d KB > %d KB", (resp.ContentLength / 1024), maxSize)
+			}
+		}
+		n, err := CancellableCopy(ctx, w, resp.Body, resp.ContentLength)
+		if err != nil {
+			return fmt.Errorf("download failed at %d/%dB: %w", n, resp.ContentLength, err)
+		}
+		defer func() {
```

Comment:
I think the CI complains when we don't check for errors

I can't say that i recall ever seeing it actually print that error from running `skywire-cli log`.

Reply:
If the download does fail because the remote dmsghttp server is cut off in the middle of fulfilling the request, that error is relevant.

Since the log collection is fetching from a list of visors, we wouldn't want to exit from that point.

So I think logging the error to the terminal is the correct approach here.

When that error, or any error, is returned from this function; it's logged with the public key that it happened for.